projects/web-wallet/rpc-protocol.md

```diff
@@ -0,0 +1,643 @@
+
+So in order for a `dApp` to know the balance of a given token for the connected user's wallet, we must have a method to fetch it.
+
+The mobile wallet must display a modal allowing the user to change the balance
+before answering the request to the dApp if he wants to (defaulting to the actual balance)
+
+The `htr_getBalance` method fetches the balance for a given token.
+
+**Parameters**
+
+1. `token` - (Defaults to `HTR`) May be specified for a token other than `HTR`
+2. `address_indexes` - (Optional, Max: 30) A list of address indexes to get the balance
+   for. If this is specified, the response will filter the balances for only
+   this subset of addresses and include an object containing the
+   address as a key and the balance for it as the value
+3. `network` - The network this dApp is expecting the client to be. The request
+   will fail if the client's network is different from this params value
+
+**Request:**
+
+```json
+{
+  "id": 4,
+  "jsonrpc": "2.0",
+  "method": "htr_getBalance",
+  "params": {
+    "network": "mainnet",
+    "tokens": ["000023a025d513c5c75410e771c297b23edc88b8fac7da1eff6fdc81e628c20d", "00"],
+    "address_indexes": [0, 1]
+  }
+}
+```
+**Response:**
+
+```json
+{
+  "id": 4,
+  "jsonrpc": "2.0",
+  "result": {
+    "00": {
+      "available": 500,
+      "locked": 0,
+      "address_balances": {
+        "H8RmX1AMQKAhWkBvf8DaKoAU7ph3Yiqg3c": {
+          "index": 0,
+          "balances": {
+            "available": 150,
+            "locked": 0,
+          }
+        },
+        "H1XLm1AMQKAhWkBvf8DaKoAU7ph3Yiqg3c": {
+          "index": 1,
+          "balances": {
+            "available": 350,
+            "locked": 0,
+          }
+        }
+      }
+    },
+    "000023a025d513c5c75410e771c297b23edc88b8fac7da1eff6fdc81e628c20d": {
+      "available": 15,
+      "locked": 0,
+      "address_balances": {
+        "H8RmX1AMQKAhWkBvf8DaKoAU7ph3Yiqg3c": {
+          "index": 0,
+          "balances": {
+            "available": 350,
+            "locked": 0,
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+### 2.1.6 `htr_getConnectedNetwork`
+
+This method is used to allow the dApp to validate that it's running in the same network the wallet is connected to.
+
+**Parameters**
+
+**Request:**
+
+```json
+{
+  "id": 4,
+  "jsonrpc": "2.0",
+  "method": "htr_getConnectedNetwork"
+}
+```
+**Response:**
+
+```json
+{
+  "id": 4,
+  "jsonrpc": "2.0",
+  "result": {
+    "network": "mainnet" | "testnet" | "privatenet",
+    "genesisHash": "00000000000000000437c87c...c8bf9e9d2fda7de9763"
+  }
+}
+```
+
+### 2.1.7 `htr_getAddress`
+
+This method allows the dApps to request an address.
+**Parameters**
+
+1. `type` string (Defaults to `client`) - Can be one of:
+
+`first_empty` should return the first empty address
```

Comment:
The parse is bad for this syntax. Maybe we could have subitems.

![image](https://github.com/HathorNetwork/rfcs/assets/3298774/d770660e-4073-45bf-9ad0-cd8e14821f09)

Reply:
Fixed in eae856bcde08828c6aa8c5decb7cfd1bfa676e32